### PR TITLE
Refactor(eos_cli_config_gen, eos_designs): Improve BGP VRF Address Families Model

### LINKED ISSUE #1323: Revisit eos_cli_config_gen BGP data model
### Enhancement summary

Refactor of router_bgp data model to be consistent for global and discrete VRF, as raised by @c-po 

```yaml
router_bgp:
  as: "100"
  address_family_ipv4:
    networks:
      10.0.0.0/8:
    neighbors:
      192.168.6.34:
        prefix_list_in: "wan-in"

  vrfs:
    FooMPLS:
      address_families:
        ipv4:
          networks:
            192.168.72.0/29:
          neighbors:
            192.168.82.72:
              activate: true
```

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

Refactor router_bgp data model - TBD.

### Describe the solution you would like

Refactor data model - TBD.

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by emilarista**
## Change Summary

* Adds new and better data model for address family configuration under router bgp vrfs.
  * ipv4
  * ipv4 multicast
  * ipv6 
  * ipv6 multicast
  * flow-spec ipv4
  * flow-spec ipv6
* Refactor eos_designs to use the new model. 
* Adds deprecation warnings for the old data model.
* Update molecule scenarios.

## Related Issue(s)

Fixes #1323 

## Component(s) name

`arista.avd.eos_cli_config_gen`
`arista.avd.eos_designs`

## Proposed changes

```yaml
    router_bgp:
      vrfs:
        - name: <str>
          ...
          address_family_ipv4:
            bgp:
              missing_policy:
                direction_in_action: <str>
                direction_out_action: <str>
              additional_paths:
                install: <bool>
                install_ecmp_primary: <bool>
                receive: <bool>
                send:
                  any: <bool>
                  backup: <bool>
                  ecmp: <bool>
                  ecmp_limit: <int>
                  limit: <int>
            neighbors:
              - ip_address: <str>
                activate: <bool>
                route_map_in: <str>
                route_map_out: <str>
            peer_groups:
              - name: <str>
                activate: <bool>
                next_hop:
                  address_family_ipv6:
                    enabled: <bool>
                    originate: <bool>
            networks:
              - prefix: <str>
                route_map: <str>
          address_family_ipv6:
            bgp:
              missing_policy:
                direction_in_action: <str>
                direction_out_action: <str>
              additional_paths:
                install: <bool>
                install_ecmp_primary: <bool>
                receive: <bool>
                send:
                  any: <bool>
                  backup: <bool>
                  ecmp: <bool>
                  ecmp_limit: <int>
                  limit: <int>
            neighbors:
              - ip_address: <str>
                activate: <bool>
                route_map_in: <str>
                route_map_out: <str>
            peer_groups:
              - name: <str>
                activate: <bool>
            networks:
              - prefix: <str>
                route_map: <str>
          address_family_ipv4_multicast:
            bgp:
              missing_policy:
                direction_in_action: <str>
                direction_out_action: <str>
              additional_paths:
                receive: <bool>
            neighbors:
              - ip_address: <str>
                activate: <bool>
                route_map_in: <str>
                route_map_out: <str>
            peer_groups:
              - name: <str>
                activate: <bool>
            networks:
              - prefix: <str>
                route_map: <str>
          address_family_ipv6_multicast:
            bgp:
              missing_policy:
                direction_in_action: <str>
                direction_out_action: <str>
              additional_paths:
                receive: <bool>
            neighbors:
              - ip_address: <str>
                activate: <bool>
                route_map_in: <str>
                route_map_out: <str>
            peer_groups:
              - name: <str>
                activate: <bool>
            networks:
              - prefix: <str>
                route_map: <str>
          address_family_flow_spec_ipv4:
            bgp:
              missing_policy:
                direction_in_action: <str>
                direction_out_action: <str>
            neighbors:
              - ip_address: <str>
                activate: <bool>
          address_family_flow_spec_ipv6:
            bgp:
              missing_policy:
                direction_in_action: <str>
                direction_out_action: <str>
            neighbors:
              - ip_address: <str>
                activate: <bool>
```

## How to test
Tested with molecule to make sure identical config is rendered.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2 [hunk 1]
```diff
@@ -1048,6 +1048,244 @@ router bgp {{ router_bgp.as }}
...
+{%                 set network_cli = "network " ~ network.prefix %}
+{%                 if network.route_map is arista.avd.defined %}
+{%                     set network_cli = network_cli ~ " route-map " ~ network.route_map %}
+{%                 endif %}
+         {{ network_cli }}
+{%             endfor %}
+{%         endif %}
+{%         if vrf.address_family_ipv6_multicast is arista.avd.defined %}
+      !
+      address-family ipv6 multicast
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:1260]: have you checked the order of these address families vs. order on EOS?

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml [hunk 2]
```diff
@@ -1474,11 +1474,426 @@ keys:
...
+                      activate:
+                        type: bool
             address_families:
               type: list
               primary_key: address_family
               convert_types:
                 - dict
+              deprecation:
+                warning: true
+                new_key: address_family_ipv4/address_family_ipv6
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml]: maybe just write `address_family_*`?

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2 [hunk 3]
```diff
@@ -1048,6 +1048,244 @@ router bgp {{ router_bgp.as }}
...
+{%             elif vrf.address_family_ipv6.bgp.additional_paths.send.limit is arista.avd.defined %}
+         bgp additional-paths send limit {{ vrf.address_family_ipv6.bgp.additional_paths.send.limit }}
+{%             endif %}
+{%             for peer_group in vrf.address_family_ipv6.peer_groups | arista.avd.natural_sort('name') %}
+{%                 if peer_group.activate is arista.avd.defined(true) %}
+         neighbor {{ peer_group.name }} activate
+{%                 elif peer_group.activate is arista.avd.defined(false) %}
+         no neighbor {{ peer_group.name }} activate
+{%                 endif %}
+{%                 if peer_group.next_hop.address_family_ipv6_originate is arista.avd.defined(true) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2]: I don't find this in the schema.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2 (hunk 1 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2:1260]: Yep, have checked and they are the same.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml (hunk 2 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml]: Fixed!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2 (hunk 3 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2]: Copy-paste error, it's removed now. Thanks!

File: ansible_collections/arista/avd/docs/porting-guides/4.x.x.md [hunk 4]
```diff
@@ -463,6 +463,68 @@ vlan_interfaces:
...
+            activate: true
+            next_hop:
+              address_family_ipv6:
+                enabled: true
+                originate: true
+      address_family_ipv6:
+        ...
+```
+
+The change has been incorporated into `eos_designs` so no action is necessary to retain the same functionality. If using `structured_configuration` to add additional configuration options under the address families using the old data models, migrating to using the new data models is recommended.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/porting-guides/4.x.x.md]:
```suggestion
The change has been incorporated into `eos_designs` so no action is necessary to retain the same functionality.

It is recommended to migrate to the new data models when using `structured_config` or `custom_structured_configuration_` to add additional configuration options under the VRF address families.
```

**Review by ClausHolbechArista — APPROVED** [review]

**Review by ClausHolbechArista — APPROVED** [review]

**Review by ClausHolbechArista — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Pull request merged**